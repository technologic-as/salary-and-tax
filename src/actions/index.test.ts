import configureMockStore from 'redux-mock-store';
import { formChange, inputHasChanged } from './index';

const mockStore = configureMockStore();
const values = { newValue: 'this is new', otherValue: 'also new' };

describe('actions', () => {
  describe('formChange', () => {
    it('should create action matching snapshot', () => {
      expect(formChange(values)).toMatchSnapshot();
    });
  });
  describe('inputHasChanged', () => {
    it('should dispatch formChange', async () => {
      const store = mockStore();
      await inputHasChanged(values)(store.dispatch);
      expect(store.getActions()).toEqual([{ type: 'FORM_CHANGE', ...values }]);
    });
  });
});
