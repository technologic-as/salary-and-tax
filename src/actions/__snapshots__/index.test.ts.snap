// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions formChange should create action matching snapshot 1`] = `
Object {
  "newValue": "this is new",
  "otherValue": "also new",
  "type": "FORM_CHANGE",
}
`;
